showcase/src/components/ui/dictation-button.tsx

```diff
@@ -0,0 +1,72 @@
+import { Tooltip } from "@/components/ui/suggestions-tooltip";
+import { useTamboThreadInput, useTamboVoice } from "@tambo-ai/react";
+import { Loader2Icon, Mic, Square } from "lucide-react";
+import { useEffect, useState } from "react";
+
+/**
+ * Button for dictating speech into the message input.
+ */
+export default function DictationButton() {
+  const {
+    startRecording,
+    stopRecording,
+    isRecording,
+    isTranscribing,
+    transcript,
+    transcriptionError,
+  } = useTamboVoice();
+  const { value, setValue } = useTamboThreadInput();
+  const [lastProcessedTranscript, setLastProcessedTranscript] =
+    useState<string>("");
+
+  const handleStartRecording = () => {
+    setLastProcessedTranscript("");
+    startRecording();
+  };
+
+  const handleStopRecording = () => {
+    stopRecording();
+  };
+
+  useEffect(() => {
+    if (transcript && transcript !== lastProcessedTranscript) {
+      setLastProcessedTranscript(transcript);
+      setValue(value + " " + transcript);
+    }
+  }, [transcript, lastProcessedTranscript, value, setValue]);
+
+  if (isTranscribing) {
+    return (
+      <div className="p-2 rounded-md">
+        <Loader2Icon className="h-5 w-5 animate-spin" />
+      </div>
+    );
+  }
+
+  if (isRecording) {
+    return (
```

Comment:
feels strange to do an early return here - because you're switching from a `<button>` to a `<div><span><button>` depending on recording state - layout may be ugly or get formatted differently. 

maybe conditionalize the <button> instead of early return?